### PR TITLE
Njb/add typings

### PR DESCRIPTION
**Pull request by NickBolles**
Add typescript typings, makes writing typescript tests much easier!

### CODE REVIEW

**Comment by NickBolles** [comment]
@bahmutov could you take a look at this?

File: index.d.ts [hunk 1]
```diff
@@ -0,0 +1,32 @@
+import Vue, { ComponentOptions } from "vue";
+
+declare global {
+  namespace Cypress {
+    import Vue from "vue";
```
Comment (bahmutov) [inline index.d.ts:5]: why the second `import` of Vue?

Comment (NickBolles) [reply index.d.ts]: Resolved. Not sure this did anything, but it works just fine without it.

**Comment by bahmutov** [comment]
Hey @NickBolles I have added TS in https://github.com/bahmutov/cypress-vue-unit-test/releases/tag/v3.4.0 I think this should be enough for now

**Comment by NickBolles** [comment]
@bahmutov Sweet! I just tried it out and there are a few things missing. 

First off, `Cypress.vue` isn't defined, unless that's changed recently, we should add this to the typings:
```typescript
declare global {
  // eslint-disable-next-line no-redeclare
  namespace Cypress {
    interface Cypress {
      vue: Vue;
    }
  }
}
```
I'm also seeing this (which looks like it should be valid based off of the readme)
image

This probably means we need to update 
to be 

```
Vue.VueConstructor<V>Vue.VueConstructor<V>

**Comment by bahmutov** [comment]
@NickBolles I have added those 2 things in https://github.com/bahmutov/cypress-vue-unit-test/pull/354 Mind taking a look? I need a good Vue app with TS setup to really test it unfortunately

**Comment by NickBolles** [comment]
@bahmutov that looks good. I can't figure out how to pull it (I thought `yarn add github:buhmutov/cypress-vue-unit-test#extend-vue-type` would do it, but I guess not?)

**Comment by bahmutov** [comment]
Check the spelling ;)

**Comment by NickBolles** [comment]
oops, that was just a miss-type into the comment though. I tried the correct spelling too :)

**Comment by bahmutov** [comment]
@NickBolles I have released the new version and it solved my TypeScript typing for a project I scaffolded for https://github.com/bahmutov/cypress-vue-unit-test/pull/356

Speaking of which - @NickBolles how did you set up the WebPack / settings for transpile for TS? I have a fresh TS project and it keeps getting an error ...

**Comment by NickBolles** [comment]
@bahmutov perfect.

I use nuxt, so I was messing with it in [this comment](https://github.com/bahmutov/cypress-vue-unit-test/issues/200#issuecomment-568301136), which builds a webpack processor in the `ts-preporcessor` file. 

Since then I've [packaged that up](https://github.com/bahmutov/cypress-vue-unit-test/issues/200#issuecomment-610169711) so that it's easier to do for Nuxt - [cypress-nuxt](https://github.com/NickBolles/cypress-nuxt)

Because the rest of the webpack config is setup correctly, cypress-vue-unit-test just adds an entry point and it goes from that entry point.

**Comment by bahmutov** [comment]
@NickBolles you should definitely add it to https://on.cypress.io/plugins#framework-tooling!

**Comment by NickBolles** [comment]
@bahmutov sweet! I submitted a PR here: https://github.com/NickBolles/cypress-documentation/pull/1

I also updated the documentation for the more recent version of `cypress-vue-unit-test`, let me know if you have any suggestions!

**Pull request closed without merging**